src/machine.rs

```diff
@@ -1051,6 +1084,16 @@ impl<'mir, 'tcx> Machine<'mir, 'tcx> for MiriMachine<'mir, 'tcx> {
             },
             |ptr| ecx.global_base_pointer(ptr),
         )?;
+
+        if let MemoryKind::Machine(kind) = kind {
+            if kind.should_save_allocation_span() {
+                ecx.machine
+                    .allocation_spans
+                    .borrow_mut()
+                    .insert(id, (Some(ecx.machine.current_span()), None));
```

Comment:
The first component seems to always be `Some`, so it doesn't have to be an `Option`.